src/org/intellij/erlang/rebar/importWizard/ImportedOtpApp.java

```diff
@@ -130,25 +131,18 @@ private void addInfoFromRebarConfig() {
     addParseTransformsFromRebarConfig(rebarConfigPsi);
   }
 
-  private void addDependenciesFromAppFile(@NotNull VirtualFile appFile) {
-    ErlangFile appConfigPsi = ErlangTermFileUtil.createPsi(appFile);
-    List<ErlangTupleExpression> applicationDescriptors = ErlangTermFileUtil.getConfigSections(appConfigPsi, "application");
-    ErlangListExpression appAttributes = PsiTreeUtil.getChildOfType(ContainerUtil.getFirstItem(applicationDescriptors), ErlangListExpression.class);
-    ErlangTermFileUtil.processConfigSection(appAttributes, "applications", new Consumer<ErlangExpression>() {
-      @Override
-      public void consume(ErlangExpression deps) {
-        ErlangListExpression dependencyAppsList = deps instanceof ErlangListExpression ? (ErlangListExpression) deps : null;
-        if (dependencyAppsList != null) {
-          for (ErlangExpression depExpression : dependencyAppsList.getExpressionList()) {
-            ErlangQAtom depApp = PsiTreeUtil.getChildOfType(depExpression, ErlangQAtom.class);
-            ErlangAtom appNameAtom = depApp != null ? depApp.getAtom() : null;
-            if (appNameAtom != null) {
-              myDeps.add(appNameAtom.getName());
-            }
-          }
+  private void addDependenciesFromAppFile(@NotNull VirtualFile virtualAppFile) {
+    ErlangFile appFile = ErlangTermFileUtil.createPsi(virtualAppFile);
+    Collection<ErlangAppParameter> parameters = PsiTreeUtil.findChildrenOfType(appFile, ErlangAppParameter.class);
+    for (ErlangAppParameter parameter : parameters) {
+      String name = parameter.getName();
+      if ("applications".equals(name)) {
+        Collection<ErlangAtom> deps = PsiTreeUtil.findChildrenOfType(parameter.getExpression(), ErlangAtom.class);
+        for (ErlangAtom dependency : deps) {
+          myDeps.add(dependency.getName());
```

Comment:
Old version is not working now.
New version is using new structure of file and doing its job. 
This code is easy to understand. It is much clearer than it was before. What exactly do not you like?